ts/Core/Time.ts

```diff
@@ -245,18 +246,35 @@ class Time extends TimeBase {
             // Handle higher ranks. Mark new days if the time is on midnight
             // (#950, #1649, #1760, #3349). Use a reasonable dropout threshold
             // to prevent looping over dense data grouping (#6156).
-            if (unitRange <= timeUnits.hour && tickPositions.length < 10000) {
-                tickPositions.forEach((t: number): void => {
-                    if (
-                        // Speed optimization, no need to run dateFormat unless
-                        // we're on a full or half hour
-                        t % 1800000 === 0 &&
-                        // Check for local or global midnight
-                        time.dateFormat('%H%M%S%L', t) === '000000000'
-                    ) {
-                        higherRanks[t] = 'day';
-                    }
-                });
+            if (tickPositions.length < 10000) {
```

Comment:
How about extracting this method from here? To external method eg `getHigherLowerRanks(unitRange, timeUnits, tickPositions)`? For the sake of readability, and would enable unit-tests, without chart creation.